Add Emscripten port

podsvirov opened the pull request:
Suggested changes allow to build sample code with [Emscripten](https://emscripten.org/) toolchain.
Links to prebuilt samples added to [README.md](https://github.com/podsvirov/opengles3-book/tree/emscripten).
Prebuilt samples available in [gh-pages](https://github.com/podsvirov/opengles3-book/tree/gh-pages) branch (you can merge it too and change links in README.md to your repo).

Cc: @kripken @sbc100

podsvirov committed:
Add Emscripten port

podsvirov committed:
Update README.md

Add links to prebuilt samples.

sbc100 commented on Common/CMakeLists.txt:
```diff
@@ -4,8 +4,16 @@ set ( common_src Source/esShader.c
+    target_link_options( Common PUBLIC
+        "SHELL:-s MAX_WEBGL_VERSION=2" "SHELL:-s FULL_ES3=1"
+        "SHELL:-s EXPORTED_RUNTIME_METHODS='[\"ccall\"]'"
+        "SHELL:--shell-file \"${CMAKE_CURRENT_LIST_DIR}/Source/Emscripten/esUtil_Emscripten.html\"" )
```
These days we support `-sFOO=bar` so I think you can avoid the "SHELL:" thing for `-s` settings if that makes it easier?

sbc100 replied:
I think this works for `--shell-file=foo` and `--preload-file=foo` too.

podsvirov committed:
Removed unnecessary `SHELL:` prefix

It's now not needed for Emscripten toolchain
while using split options in CMake files.

podsvirov committed:
Emscripten code cleaning

podsvirov commented:
@danginsburg, what do you think of these suggestions? Is there a chance for a merger?

cmello commented:
Looks good to me, please could we merge to enable emscripten support?
